Set 'errors' attribute via kwarg in Api.init_app

robertglen opened the pull request:
required to support flask-restful extended error handling

noirbizarre commented:
Hi!
I'm not very fond of Flask-restful error handling, motly because text based approch oes not support inheritance.

In Flask-RESTPlus, the prefered approcha is using the `@api.errorhandler` decorator which is type-based and support inheritance.

A simple example with Flask-Principal PermissionDenied:
```python
@api.errorhandler(PermissionDenied)
def handle_permission_denied(error):
    message = 'You do not have the required permission to perform this action.'
    return {
        'message': message
    }, 403
```

You can return whatever you want or need, with the HTTP status code you want.

robertglen commented:
Hi Axel,

Thanks for the feedback.

I just noticed a 0.8.0  which I updated to and now I can't get to any
routes.  I am getting this traceback.


Traceback (most recent call last):
  File
"/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
line 1820, in wsgi_app
    response = self.make_response(self.handle_exception(e))
  File
"/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
line 270, in error_router
    return original_handler(e)
  File
"/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
line 270, in error_router
    return original_handler(e)
  File
"/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
line 1403, in handle_exception
    reraise(exc_type, exc_value, tb)
  File
"/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
line 267, in error_router
    return self.handle_error(e)
  File
"/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restplus/api.py",
line 379, in handle_error
    return super(Api, self).handle_error(e)
  File
"/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
line 1817, in wsgi_app
    response = self.full_dispatch_request()
  File
"/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
line 1477, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File
"/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
line 270, in error_router
    return original_handler(e)
  File
"/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
line 270, in error_router
    return original_handler(e)
  File
"/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
line 1381, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File
"/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
line 267, in error_router
    return self.handle_error(e)
  File
"/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restplus/api.py",
line 379, in handle_error
    return super(Api, self).handle_error(e)
  File
"/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
line 1475, in full_dispatch_request
    rv = self.dispatch_request()
  File
"/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
line 1461, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File
"/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
line 471, in wrapper
    resp = resource(*args, **kwargs)
  File
"/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/views.py",
line 83, in view
    self = view.view_class(*class_args, **class_kwargs)
TypeError: __init__() takes exactly 1 argument (2 given)


I spent some time trying to figure out where the issue is, but it's not
obvious to me.  When I roll the flask-restplus to 0.7.2... everything is
fine.  So something in the 0.8.0 changes are breaking everything for me.

any ideas?

On Mon, Oct 19, 2015 at 2:35 PM, Axel Haustant <notifications@github.com>
wrote:

> Hi!
> I'm not very fond of Flask-restful error handling, motly because text
> based approch oes not support inheritance.
>
> In Flask-RESTPlus, the prefered approcha is using the @api.errorhandler
> decorator which is type-based and support inheritance.
>
> A simple example with Flask-Principal PermissionDenied:
>
> @api.errorhandler(PermissionDenied)def handle_permission_denied(error):
>     message = 'You do not have the required permission to perform this action.'
>     return {
>         'message': message
>     }, 403
>
> You can return whatever you want or need, with the HTTP status code you
> want.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/noirbizarre/flask-restplus/pull/85#issuecomment-149354417>
> .
>



-- 
Robert Glen

noirbizarre commented:
I think your resources inherits from `flask.ext.restful.Resource` instead
of `flask.ext.restplus.Resource`.
Is it true?
Le 21 oct. 2015 8:49 PM, "Robert Glen" <notifications@github.com> a écrit :

> Hi Axel,
>
> Thanks for the feedback.
>
> I just noticed a 0.8.0 which I updated to and now I can't get to any
> routes. I am getting this traceback.
>
>
> Traceback (most recent call last):
> File
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
> line 1820, in wsgi_app
> response = self.make_response(self.handle_exception(e))
> File
>
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
> line 270, in error_router
> return original_handler(e)
> File
>
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
> line 270, in error_router
> return original_handler(e)
> File
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
> line 1403, in handle_exception
> reraise(exc_type, exc_value, tb)
> File
>
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
> line 267, in error_router
> return self.handle_error(e)
> File
>
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restplus/api.py",
> line 379, in handle_error
> return super(Api, self).handle_error(e)
> File
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
> line 1817, in wsgi_app
> response = self.full_dispatch_request()
> File
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
> line 1477, in full_dispatch_request
> rv = self.handle_user_exception(e)
> File
>
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
> line 270, in error_router
> return original_handler(e)
> File
>
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
> line 270, in error_router
> return original_handler(e)
> File
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
> line 1381, in handle_user_exception
> reraise(exc_type, exc_value, tb)
> File
>
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
> line 267, in error_router
> return self.handle_error(e)
> File
>
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restplus/api.py",
> line 379, in handle_error
> return super(Api, self).handle_error(e)
> File
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
> line 1475, in full_dispatch_request
> rv = self.dispatch_request()
> File
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
> line 1461, in dispatch_request
> return self.view_functions[rule.endpoint](**req.view_args)
> File
>
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
> line 471, in wrapper
> resp = resource(*args, **kwargs)
> File
>
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/views.py",
> line 83, in view
> self = view.view_class(*class_args, **class_kwargs)
> TypeError: __init__() takes exactly 1 argument (2 given)
>
>
> I spent some time trying to figure out where the issue is, but it's not
> obvious to me. When I roll the flask-restplus to 0.7.2... everything is
> fine. So something in the 0.8.0 changes are breaking everything for me.
>
> any ideas?
>
> On Mon, Oct 19, 2015 at 2:35 PM, Axel Haustant <notifications@github.com>
> wrote:
>
> > Hi!
> > I'm not very fond of Flask-restful error handling, motly because text
> > based approch oes not support inheritance.
> >
> > In Flask-RESTPlus, the prefered approcha is using the @api.errorhandler
> > decorator which is type-based and support inheritance.
> >
> > A simple example with Flask-Principal PermissionDenied:
> >
> > @api.errorhandler(PermissionDenied)def handle_permission_denied(error):
> > message = 'You do not have the required permission to perform this
> action.'
> > return {
> > 'message': message
> > }, 403
> >
> > You can return whatever you want or need, with the HTTP status code you
> > want.
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
> https://github.com/noirbizarre/flask-restplus/pull/85#issuecomment-149354417
> >
> > .
> >
>
>
>
> --
> Robert Glen
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/noirbizarre/flask-restplus/pull/85#issuecomment-149991041>
> .
>

robertglen commented:
Just checked, not true.  I am importing Resource from flask_restplus


Thanks for the super prompt response btw.  I'm not done looking into it on
my end, but it sounds like I'm the first to experience this.

On Wed, Oct 21, 2015 at 11:53 AM, Axel Haustant <notifications@github.com>
wrote:

> I think your resources inherits from `flask.ext.restful.Resource` instead
> of `flask.ext.restplus.Resource`.
> Is it true?
>
> Le 21 oct. 2015 8:49 PM, "Robert Glen" <notifications@github.com> a écrit
> :
>
> > Hi Axel,
> >
> > Thanks for the feedback.
> >
> > I just noticed a 0.8.0 which I updated to and now I can't get to any
> > routes. I am getting this traceback.
> >
> >
> > Traceback (most recent call last):
> > File
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
> > line 1820, in wsgi_app
> > response = self.make_response(self.handle_exception(e))
> > File
> >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
> > line 270, in error_router
> > return original_handler(e)
> > File
> >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
> > line 270, in error_router
> > return original_handler(e)
> > File
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
> > line 1403, in handle_exception
> > reraise(exc_type, exc_value, tb)
> > File
> >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
> > line 267, in error_router
> > return self.handle_error(e)
> > File
> >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restplus/api.py",
> > line 379, in handle_error
> > return super(Api, self).handle_error(e)
> > File
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
> > line 1817, in wsgi_app
> > response = self.full_dispatch_request()
> > File
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
> > line 1477, in full_dispatch_request
> > rv = self.handle_user_exception(e)
> > File
> >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
> > line 270, in error_router
> > return original_handler(e)
> > File
> >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
> > line 270, in error_router
> > return original_handler(e)
> > File
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
> > line 1381, in handle_user_exception
> > reraise(exc_type, exc_value, tb)
> > File
> >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
> > line 267, in error_router
> > return self.handle_error(e)
> > File
> >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restplus/api.py",
> > line 379, in handle_error
> > return super(Api, self).handle_error(e)
> > File
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
> > line 1475, in full_dispatch_request
> > rv = self.dispatch_request()
> > File
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
> > line 1461, in dispatch_request
> > return self.view_functions[rule.endpoint](**req.view_args)
> > File
> >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
> > line 471, in wrapper
> > resp = resource(*args, **kwargs)
> > File
> >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/views.py",
> > line 83, in view
> > self = view.view_class(*class_args, **class_kwargs)
> > TypeError: __init__() takes exactly 1 argument (2 given)
> >
> >
> > I spent some time trying to figure out where the issue is, but it's not
> > obvious to me. When I roll the flask-restplus to 0.7.2... everything is
> > fine. So something in the 0.8.0 changes are breaking everything for me.
> >
> > any ideas?
> >
> > On Mon, Oct 19, 2015 at 2:35 PM, Axel Haustant <notifications@github.com
> >
> > wrote:
> >
> > > Hi!
> > > I'm not very fond of Flask-restful error handling, motly because text
> > > based approch oes not support inheritance.
> > >
> > > In Flask-RESTPlus, the prefered approcha is using the @api.errorhandler
> > > decorator which is type-based and support inheritance.
> > >
> > > A simple example with Flask-Principal PermissionDenied:
> > >
> > > @api.errorhandler(PermissionDenied)def handle_permission_denied(error):
> > > message = 'You do not have the required permission to perform this
> > action.'
> > > return {
> > > 'message': message
> > > }, 403
> > >
> > > You can return whatever you want or need, with the HTTP status code you
> > > want.
> > >
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <
> >
> https://github.com/noirbizarre/flask-restplus/pull/85#issuecomment-149354417
> > >
> > > .
> > >
> >
> >
> >
> > --
> > Robert Glen
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
> https://github.com/noirbizarre/flask-restplus/pull/85#issuecomment-149991041
> >
>
> > .
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/noirbizarre/flask-restplus/pull/85#issuecomment-149992756>
> .
>



-- 
Robert Glen

noirbizarre commented:
I will look into it.
It you have some code to reproduce it...
Le 21 oct. 2015 8:58 PM, "Robert Glen" <notifications@github.com> a écrit :

> Just checked, not true. I am importing Resource from flask_restplus
>
>
> Thanks for the super prompt response btw. I'm not done looking into it on
> my end, but it sounds like I'm the first to experience this.
>
> On Wed, Oct 21, 2015 at 11:53 AM, Axel Haustant <notifications@github.com>
> wrote:
>
> > I think your resources inherits from `flask.ext.restful.Resource` instead
> > of `flask.ext.restplus.Resource`.
> > Is it true?
> >
> > Le 21 oct. 2015 8:49 PM, "Robert Glen" <notifications@github.com> a
> écrit
> > :
> >
> > > Hi Axel,
> > >
> > > Thanks for the feedback.
> > >
> > > I just noticed a 0.8.0 which I updated to and now I can't get to any
> > > routes. I am getting this traceback.
> > >
> > >
> > > Traceback (most recent call last):
> > > File
> > >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
> > > line 1820, in wsgi_app
> > > response = self.make_response(self.handle_exception(e))
> > > File
> > >
> > >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
> > > line 270, in error_router
> > > return original_handler(e)
> > > File
> > >
> > >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
> > > line 270, in error_router
> > > return original_handler(e)
> > > File
> > >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
> > > line 1403, in handle_exception
> > > reraise(exc_type, exc_value, tb)
> > > File
> > >
> > >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
> > > line 267, in error_router
> > > return self.handle_error(e)
> > > File
> > >
> > >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restplus/api.py",
> > > line 379, in handle_error
> > > return super(Api, self).handle_error(e)
> > > File
> > >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
> > > line 1817, in wsgi_app
> > > response = self.full_dispatch_request()
> > > File
> > >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
> > > line 1477, in full_dispatch_request
> > > rv = self.handle_user_exception(e)
> > > File
> > >
> > >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
> > > line 270, in error_router
> > > return original_handler(e)
> > > File
> > >
> > >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
> > > line 270, in error_router
> > > return original_handler(e)
> > > File
> > >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
> > > line 1381, in handle_user_exception
> > > reraise(exc_type, exc_value, tb)
> > > File
> > >
> > >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
> > > line 267, in error_router
> > > return self.handle_error(e)
> > > File
> > >
> > >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restplus/api.py",
> > > line 379, in handle_error
> > > return super(Api, self).handle_error(e)
> > > File
> > >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
> > > line 1475, in full_dispatch_request
> > > rv = self.dispatch_request()
> > > File
> > >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
> > > line 1461, in dispatch_request
> > > return self.view_functions[rule.endpoint](**req.view_args)
> > > File
> > >
> > >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
> > > line 471, in wrapper
> > > resp = resource(*args, **kwargs)
> > > File
> > >
> > >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/views.py",
> > > line 83, in view
> > > self = view.view_class(*class_args, **class_kwargs)
> > > TypeError: __init__() takes exactly 1 argument (2 given)
> > >
> > >
> > > I spent some time trying to figure out where the issue is, but it's not
> > > obvious to me. When I roll the flask-restplus to 0.7.2... everything is
> > > fine. So something in the 0.8.0 changes are breaking everything for me.
> > >
> > > any ideas?
> > >
> > > On Mon, Oct 19, 2015 at 2:35 PM, Axel Haustant <
> notifications@github.com
> > >
> > > wrote:
> > >
> > > > Hi!
> > > > I'm not very fond of Flask-restful error handling, motly because text
> > > > based approch oes not support inheritance.
> > > >
> > > > In Flask-RESTPlus, the prefered approcha is using the
> @api.errorhandler
> > > > decorator which is type-based and support inheritance.
> > > >
> > > > A simple example with Flask-Principal PermissionDenied:
> > > >
> > > > @api.errorhandler(PermissionDenied)def
> handle_permission_denied(error):
> > > > message = 'You do not have the required permission to perform this
> > > action.'
> > > > return {
> > > > 'message': message
> > > > }, 403
> > > >
> > > > You can return whatever you want or need, with the HTTP status code
> you
> > > > want.
> > > >
> > > > —
> > > > Reply to this email directly or view it on GitHub
> > > > <
> > >
> >
> https://github.com/noirbizarre/flask-restplus/pull/85#issuecomment-149354417
> > > >
> > > > .
> > > >
> > >
> > >
> > >
> > > --
> > > Robert Glen
> > >
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <
> >
> https://github.com/noirbizarre/flask-restplus/pull/85#issuecomment-149991041
> > >
> >
> > > .
> > >
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
> https://github.com/noirbizarre/flask-restplus/pull/85#issuecomment-149992756
> >
> > .
> >
>
>
>
> --
> Robert Glen
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/noirbizarre/flask-restplus/pull/85#issuecomment-149993922>
> .
>

robertglen commented:
I'll see if I can create a dummy resource endpoint with a GET and then
figure out what needs to be shared.

On Wed, Oct 21, 2015 at 12:01 PM, Axel Haustant <notifications@github.com>
wrote:

> I will look into it.
> It you have some code to reproduce it...
>
> Le 21 oct. 2015 8:58 PM, "Robert Glen" <notifications@github.com> a écrit
> :
>
> > Just checked, not true. I am importing Resource from flask_restplus
> >
> >
> > Thanks for the super prompt response btw. I'm not done looking into it on
> > my end, but it sounds like I'm the first to experience this.
> >
> > On Wed, Oct 21, 2015 at 11:53 AM, Axel Haustant <
> notifications@github.com>
> > wrote:
> >
> > > I think your resources inherits from `flask.ext.restful.Resource`
> instead
> > > of `flask.ext.restplus.Resource`.
> > > Is it true?
> > >
> > > Le 21 oct. 2015 8:49 PM, "Robert Glen" <notifications@github.com> a
> > écrit
> > > :
> > >
> > > > Hi Axel,
> > > >
> > > > Thanks for the feedback.
> > > >
> > > > I just noticed a 0.8.0 which I updated to and now I can't get to any
> > > > routes. I am getting this traceback.
> > > >
> > > >
> > > > Traceback (most recent call last):
> > > > File
> > > >
> > >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
> > > > line 1820, in wsgi_app
> > > > response = self.make_response(self.handle_exception(e))
> > > > File
> > > >
> > > >
> > >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
> > > > line 270, in error_router
> > > > return original_handler(e)
> > > > File
> > > >
> > > >
> > >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
> > > > line 270, in error_router
> > > > return original_handler(e)
> > > > File
> > > >
> > >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
> > > > line 1403, in handle_exception
> > > > reraise(exc_type, exc_value, tb)
> > > > File
> > > >
> > > >
> > >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
> > > > line 267, in error_router
> > > > return self.handle_error(e)
> > > > File
> > > >
> > > >
> > >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restplus/api.py",
> > > > line 379, in handle_error
> > > > return super(Api, self).handle_error(e)
> > > > File
> > > >
> > >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
> > > > line 1817, in wsgi_app
> > > > response = self.full_dispatch_request()
> > > > File
> > > >
> > >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
> > > > line 1477, in full_dispatch_request
> > > > rv = self.handle_user_exception(e)
> > > > File
> > > >
> > > >
> > >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
> > > > line 270, in error_router
> > > > return original_handler(e)
> > > > File
> > > >
> > > >
> > >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
> > > > line 270, in error_router
> > > > return original_handler(e)
> > > > File
> > > >
> > >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
> > > > line 1381, in handle_user_exception
> > > > reraise(exc_type, exc_value, tb)
> > > > File
> > > >
> > > >
> > >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
> > > > line 267, in error_router
> > > > return self.handle_error(e)
> > > > File
> > > >
> > > >
> > >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restplus/api.py",
> > > > line 379, in handle_error
> > > > return super(Api, self).handle_error(e)
> > > > File
> > > >
> > >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
> > > > line 1475, in full_dispatch_request
> > > > rv = self.dispatch_request()
> > > > File
> > > >
> > >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
> > > > line 1461, in dispatch_request
> > > > return self.view_functions[rule.endpoint](**req.view_args)
> > > > File
> > > >
> > > >
> > >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
> > > > line 471, in wrapper
> > > > resp = resource(*args, **kwargs)
> > > > File
> > > >
> > > >
> > >
> >
> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/views.py",
> > > > line 83, in view
> > > > self = view.view_class(*class_args, **class_kwargs)
> > > > TypeError: __init__() takes exactly 1 argument (2 given)
> > > >
> > > >
> > > > I spent some time trying to figure out where the issue is, but it's
> not
> > > > obvious to me. When I roll the flask-restplus to 0.7.2... everything
> is
> > > > fine. So something in the 0.8.0 changes are breaking everything for
> me.
> > > >
> > > > any ideas?
> > > >
> > > > On Mon, Oct 19, 2015 at 2:35 PM, Axel Haustant <
> > notifications@github.com
> > > >
> > > > wrote:
> > > >
> > > > > Hi!
> > > > > I'm not very fond of Flask-restful error handling, motly because
> text
> > > > > based approch oes not support inheritance.
> > > > >
> > > > > In Flask-RESTPlus, the prefered approcha is using the
> > @api.errorhandler
> > > > > decorator which is type-based and support inheritance.
> > > > >
> > > > > A simple example with Flask-Principal PermissionDenied:
> > > > >
> > > > > @api.errorhandler(PermissionDenied)def
> > handle_permission_denied(error):
> > > > > message = 'You do not have the required permission to perform this
> > > > action.'
> > > > > return {
> > > > > 'message': message
> > > > > }, 403
> > > > >
> > > > > You can return whatever you want or need, with the HTTP status code
> > you
> > > > > want.
> > > > >
> > > > > —
> > > > > Reply to this email directly or view it on GitHub
> > > > > <
> > > >
> > >
> >
> https://github.com/noirbizarre/flask-restplus/pull/85#issuecomment-149354417
> > > > >
> > > > > .
> > > > >
> > > >
> > > >
> > > >
> > > > --
> > > > Robert Glen
> > > >
> > > > —
> > > > Reply to this email directly or view it on GitHub
> > > > <
> > >
> >
> https://github.com/noirbizarre/flask-restplus/pull/85#issuecomment-149991041
> > > >
> > >
> > > > .
> > > >
> > >
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <
> >
> https://github.com/noirbizarre/flask-restplus/pull/85#issuecomment-149992756
> > >
> > > .
> > >
> >
> >
> >
> > --
> > Robert Glen
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
> https://github.com/noirbizarre/flask-restplus/pull/85#issuecomment-149993922
> >
>
> > .
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/noirbizarre/flask-restplus/pull/85#issuecomment-149994967>
> .
>



-- 
Robert Glen

robertglen commented:
Hi, it's been a busy day, but the little bit I've put into this so far has
shown that it does NOT happen when I use the internal python webserver that
you get with flask-script's runserver argument.

I DO get the error when uwsgi loads the app.  I literally launched my
./manage.py runserver (which listens on a different port than nginx) and
when I hit the python webserver... success,   but when I hit the normal
port hosted by nginx -> uwsgi... fail..

Maybe try whatever you guys have fronted by uwsgi and see if you can
reproduce?   Otherwise I'll continue stripping out a bare bones app and
send to you.

On Wed, Oct 21, 2015 at 12:05 PM, Robert Glen <robert.glen@gmail.com> wrote:

> I'll see if I can create a dummy resource endpoint with a GET and then
> figure out what needs to be shared.
>
> On Wed, Oct 21, 2015 at 12:01 PM, Axel Haustant <notifications@github.com>
> wrote:
>
>> I will look into it.
>> It you have some code to reproduce it...
>>
>> Le 21 oct. 2015 8:58 PM, "Robert Glen" <notifications@github.com> a
>> écrit :
>>
>> > Just checked, not true. I am importing Resource from flask_restplus
>> >
>> >
>> > Thanks for the super prompt response btw. I'm not done looking into it
>> on
>> > my end, but it sounds like I'm the first to experience this.
>> >
>> > On Wed, Oct 21, 2015 at 11:53 AM, Axel Haustant <
>> notifications@github.com>
>> > wrote:
>> >
>> > > I think your resources inherits from `flask.ext.restful.Resource`
>> instead
>> > > of `flask.ext.restplus.Resource`.
>> > > Is it true?
>> > >
>> > > Le 21 oct. 2015 8:49 PM, "Robert Glen" <notifications@github.com> a
>> > écrit
>> > > :
>> > >
>> > > > Hi Axel,
>> > > >
>> > > > Thanks for the feedback.
>> > > >
>> > > > I just noticed a 0.8.0 which I updated to and now I can't get to any
>> > > > routes. I am getting this traceback.
>> > > >
>> > > >
>> > > > Traceback (most recent call last):
>> > > > File
>> > > >
>> > >
>> >
>> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
>> > > > line 1820, in wsgi_app
>> > > > response = self.make_response(self.handle_exception(e))
>> > > > File
>> > > >
>> > > >
>> > >
>> >
>> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
>> > > > line 270, in error_router
>> > > > return original_handler(e)
>> > > > File
>> > > >
>> > > >
>> > >
>> >
>> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
>> > > > line 270, in error_router
>> > > > return original_handler(e)
>> > > > File
>> > > >
>> > >
>> >
>> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
>> > > > line 1403, in handle_exception
>> > > > reraise(exc_type, exc_value, tb)
>> > > > File
>> > > >
>> > > >
>> > >
>> >
>> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
>> > > > line 267, in error_router
>> > > > return self.handle_error(e)
>> > > > File
>> > > >
>> > > >
>> > >
>> >
>> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restplus/api.py",
>> > > > line 379, in handle_error
>> > > > return super(Api, self).handle_error(e)
>> > > > File
>> > > >
>> > >
>> >
>> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
>> > > > line 1817, in wsgi_app
>> > > > response = self.full_dispatch_request()
>> > > > File
>> > > >
>> > >
>> >
>> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
>> > > > line 1477, in full_dispatch_request
>> > > > rv = self.handle_user_exception(e)
>> > > > File
>> > > >
>> > > >
>> > >
>> >
>> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
>> > > > line 270, in error_router
>> > > > return original_handler(e)
>> > > > File
>> > > >
>> > > >
>> > >
>> >
>> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
>> > > > line 270, in error_router
>> > > > return original_handler(e)
>> > > > File
>> > > >
>> > >
>> >
>> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
>> > > > line 1381, in handle_user_exception
>> > > > reraise(exc_type, exc_value, tb)
>> > > > File
>> > > >
>> > > >
>> > >
>> >
>> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
>> > > > line 267, in error_router
>> > > > return self.handle_error(e)
>> > > > File
>> > > >
>> > > >
>> > >
>> >
>> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restplus/api.py",
>> > > > line 379, in handle_error
>> > > > return super(Api, self).handle_error(e)
>> > > > File
>> > > >
>> > >
>> >
>> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
>> > > > line 1475, in full_dispatch_request
>> > > > rv = self.dispatch_request()
>> > > > File
>> > > >
>> > >
>> >
>> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
>> > > > line 1461, in dispatch_request
>> > > > return self.view_functions[rule.endpoint](**req.view_args)
>> > > > File
>> > > >
>> > > >
>> > >
>> >
>> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
>> > > > line 471, in wrapper
>> > > > resp = resource(*args, **kwargs)
>> > > > File
>> > > >
>> > > >
>> > >
>> >
>> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/views.py",
>> > > > line 83, in view
>> > > > self = view.view_class(*class_args, **class_kwargs)
>> > > > TypeError: __init__() takes exactly 1 argument (2 given)
>> > > >
>> > > >
>> > > > I spent some time trying to figure out where the issue is, but it's
>> not
>> > > > obvious to me. When I roll the flask-restplus to 0.7.2...
>> everything is
>> > > > fine. So something in the 0.8.0 changes are breaking everything for
>> me.
>> > > >
>> > > > any ideas?
>> > > >
>> > > > On Mon, Oct 19, 2015 at 2:35 PM, Axel Haustant <
>> > notifications@github.com
>> > > >
>> > > > wrote:
>> > > >
>> > > > > Hi!
>> > > > > I'm not very fond of Flask-restful error handling, motly because
>> text
>> > > > > based approch oes not support inheritance.
>> > > > >
>> > > > > In Flask-RESTPlus, the prefered approcha is using the
>> > @api.errorhandler
>> > > > > decorator which is type-based and support inheritance.
>> > > > >
>> > > > > A simple example with Flask-Principal PermissionDenied:
>> > > > >
>> > > > > @api.errorhandler(PermissionDenied)def
>> > handle_permission_denied(error):
>> > > > > message = 'You do not have the required permission to perform this
>> > > > action.'
>> > > > > return {
>> > > > > 'message': message
>> > > > > }, 403
>> > > > >
>> > > > > You can return whatever you want or need, with the HTTP status
>> code
>> > you
>> > > > > want.
>> > > > >
>> > > > > —
>> > > > > Reply to this email directly or view it on GitHub
>> > > > > <
>> > > >
>> > >
>> >
>> https://github.com/noirbizarre/flask-restplus/pull/85#issuecomment-149354417
>> > > > >
>> > > > > .
>> > > > >
>> > > >
>> > > >
>> > > >
>> > > > --
>> > > > Robert Glen
>> > > >
>> > > > —
>> > > > Reply to this email directly or view it on GitHub
>> > > > <
>> > >
>> >
>> https://github.com/noirbizarre/flask-restplus/pull/85#issuecomment-149991041
>> > > >
>> > >
>> > > > .
>> > > >
>> > >
>> > > —
>> > > Reply to this email directly or view it on GitHub
>> > > <
>> >
>> https://github.com/noirbizarre/flask-restplus/pull/85#issuecomment-149992756
>> > >
>> > > .
>> > >
>> >
>> >
>> >
>> > --
>> > Robert Glen
>> >
>> > —
>> > Reply to this email directly or view it on GitHub
>> > <
>> https://github.com/noirbizarre/flask-restplus/pull/85#issuecomment-149993922
>> >
>>
>> > .
>> >
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/noirbizarre/flask-restplus/pull/85#issuecomment-149994967>
>> .
>>
>
>
>
> --
> Robert Glen
>



-- 
Robert Glen

robertglen commented:
Okay, Please ignore my comment about it only happening in uwsgi.  I didn't
realize I had rolled the flask-restplus library back to 0.7.2 during that
test.

I've attached a tarball that you should be able to extract, and run with
./manage.py runserver

Then hit the server on :8081/dummy/api/dummy/test

I get the traceback on 0.8.0 of flask-restplus.   When i roll back to 0.7.2
  it's fine.

On Wed, Oct 21, 2015 at 4:01 PM, Robert Glen <robert.glen@gmail.com> wrote:

> Hi, it's been a busy day, but the little bit I've put into this so far has
> shown that it does NOT happen when I use the internal python webserver that
> you get with flask-script's runserver argument.
>
> I DO get the error when uwsgi loads the app.  I literally launched my
> ./manage.py runserver (which listens on a different port than nginx) and
> when I hit the python webserver... success,   but when I hit the normal
> port hosted by nginx -> uwsgi... fail..
>
> Maybe try whatever you guys have fronted by uwsgi and see if you can
> reproduce?   Otherwise I'll continue stripping out a bare bones app and
> send to you.
>
> On Wed, Oct 21, 2015 at 12:05 PM, Robert Glen <robert.glen@gmail.com>
> wrote:
>
>> I'll see if I can create a dummy resource endpoint with a GET and then
>> figure out what needs to be shared.
>>
>> On Wed, Oct 21, 2015 at 12:01 PM, Axel Haustant <notifications@github.com
>> > wrote:
>>
>>> I will look into it.
>>> It you have some code to reproduce it...
>>>
>>> Le 21 oct. 2015 8:58 PM, "Robert Glen" <notifications@github.com> a
>>> écrit :
>>>
>>> > Just checked, not true. I am importing Resource from flask_restplus
>>> >
>>> >
>>> > Thanks for the super prompt response btw. I'm not done looking into it
>>> on
>>> > my end, but it sounds like I'm the first to experience this.
>>> >
>>> > On Wed, Oct 21, 2015 at 11:53 AM, Axel Haustant <
>>> notifications@github.com>
>>> > wrote:
>>> >
>>> > > I think your resources inherits from `flask.ext.restful.Resource`
>>> instead
>>> > > of `flask.ext.restplus.Resource`.
>>> > > Is it true?
>>> > >
>>> > > Le 21 oct. 2015 8:49 PM, "Robert Glen" <notifications@github.com> a
>>> > écrit
>>> > > :
>>> > >
>>> > > > Hi Axel,
>>> > > >
>>> > > > Thanks for the feedback.
>>> > > >
>>> > > > I just noticed a 0.8.0 which I updated to and now I can't get to
>>> any
>>> > > > routes. I am getting this traceback.
>>> > > >
>>> > > >
>>> > > > Traceback (most recent call last):
>>> > > > File
>>> > > >
>>> > >
>>> >
>>> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
>>> > > > line 1820, in wsgi_app
>>> > > > response = self.make_response(self.handle_exception(e))
>>> > > > File
>>> > > >
>>> > > >
>>> > >
>>> >
>>> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
>>> > > > line 270, in error_router
>>> > > > return original_handler(e)
>>> > > > File
>>> > > >
>>> > > >
>>> > >
>>> >
>>> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
>>> > > > line 270, in error_router
>>> > > > return original_handler(e)
>>> > > > File
>>> > > >
>>> > >
>>> >
>>> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
>>> > > > line 1403, in handle_exception
>>> > > > reraise(exc_type, exc_value, tb)
>>> > > > File
>>> > > >
>>> > > >
>>> > >
>>> >
>>> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
>>> > > > line 267, in error_router
>>> > > > return self.handle_error(e)
>>> > > > File
>>> > > >
>>> > > >
>>> > >
>>> >
>>> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restplus/api.py",
>>> > > > line 379, in handle_error
>>> > > > return super(Api, self).handle_error(e)
>>> > > > File
>>> > > >
>>> > >
>>> >
>>> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
>>> > > > line 1817, in wsgi_app
>>> > > > response = self.full_dispatch_request()
>>> > > > File
>>> > > >
>>> > >
>>> >
>>> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
>>> > > > line 1477, in full_dispatch_request
>>> > > > rv = self.handle_user_exception(e)
>>> > > > File
>>> > > >
>>> > > >
>>> > >
>>> >
>>> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
>>> > > > line 270, in error_router
>>> > > > return original_handler(e)
>>> > > > File
>>> > > >
>>> > > >
>>> > >
>>> >
>>> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
>>> > > > line 270, in error_router
>>> > > > return original_handler(e)
>>> > > > File
>>> > > >
>>> > >
>>> >
>>> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
>>> > > > line 1381, in handle_user_exception
>>> > > > reraise(exc_type, exc_value, tb)
>>> > > > File
>>> > > >
>>> > > >
>>> > >
>>> >
>>> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
>>> > > > line 267, in error_router
>>> > > > return self.handle_error(e)
>>> > > > File
>>> > > >
>>> > > >
>>> > >
>>> >
>>> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restplus/api.py",
>>> > > > line 379, in handle_error
>>> > > > return super(Api, self).handle_error(e)
>>> > > > File
>>> > > >
>>> > >
>>> >
>>> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
>>> > > > line 1475, in full_dispatch_request
>>> > > > rv = self.dispatch_request()
>>> > > > File
>>> > > >
>>> > >
>>> >
>>> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/app.py",
>>> > > > line 1461, in dispatch_request
>>> > > > return self.view_functions[rule.endpoint](**req.view_args)
>>> > > > File
>>> > > >
>>> > > >
>>> > >
>>> >
>>> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask_restful/__init__.py",
>>> > > > line 471, in wrapper
>>> > > > resp = resource(*args, **kwargs)
>>> > > > File
>>> > > >
>>> > > >
>>> > >
>>> >
>>> "/opt/crowdstrike/venvs/infraapi/lib/python2.7/site-packages/flask/views.py",
>>> > > > line 83, in view
>>> > > > self = view.view_class(*class_args, **class_kwargs)
>>> > > > TypeError: __init__() takes exactly 1 argument (2 given)
>>> > > >
>>> > > >
>>> > > > I spent some time trying to figure out where the issue is, but
>>> it's not
>>> > > > obvious to me. When I roll the flask-restplus to 0.7.2...
>>> everything is
>>> > > > fine. So something in the 0.8.0 changes are breaking everything
>>> for me.
>>> > > >
>>> > > > any ideas?
>>> > > >
>>> > > > On Mon, Oct 19, 2015 at 2:35 PM, Axel Haustant <
>>> > notifications@github.com
>>> > > >
>>> > > > wrote:
>>> > > >
>>> > > > > Hi!
>>> > > > > I'm not very fond of Flask-restful error handling, motly because
>>> text
>>> > > > > based approch oes not support inheritance.
>>> > > > >
>>> > > > > In Flask-RESTPlus, the prefered approcha is using the
>>> > @api.errorhandler
>>> > > > > decorator which is type-based and support inheritance.
>>> > > > >
>>> > > > > A simple example with Flask-Principal PermissionDenied:
>>> > > > >
>>> > > > > @api.errorhandler(PermissionDenied)def
>>> > handle_permission_denied(error):
>>> > > > > message = 'You do not have the required permission to perform
>>> this
>>> > > > action.'
>>> > > > > return {
>>> > > > > 'message': message
>>> > > > > }, 403
>>> > > > >
>>> > > > > You can return whatever you want or need, with the HTTP status
>>> code
>>> > you
>>> > > > > want.
>>> > > > >
>>> > > > > —
>>> > > > > Reply to this email directly or view it on GitHub
>>> > > > > <
>>> > > >
>>> > >
>>> >
>>> https://github.com/noirbizarre/flask-restplus/pull/85#issuecomment-149354417
>>> > > > >
>>> > > > > .
>>> > > > >
>>> > > >
>>> > > >
>>> > > >
>>> > > > --
>>> > > > Robert Glen
>>> > > >
>>> > > > —
>>> > > > Reply to this email directly or view it on GitHub
>>> > > > <
>>> > >
>>> >
>>> https://github.com/noirbizarre/flask-restplus/pull/85#issuecomment-149991041
>>> > > >
>>> > >
>>> > > > .
>>> > > >
>>> > >
>>> > > —
>>> > > Reply to this email directly or view it on GitHub
>>> > > <
>>> >
>>> https://github.com/noirbizarre/flask-restplus/pull/85#issuecomment-149992756
>>> > >
>>> > > .
>>> > >
>>> >
>>> >
>>> >
>>> > --
>>> > Robert Glen
>>> >
>>> > —
>>> > Reply to this email directly or view it on GitHub
>>> > <
>>> https://github.com/noirbizarre/flask-restplus/pull/85#issuecomment-149993922
>>> >
>>>
>>> > .
>>> >
>>>
>>> —
>>> Reply to this email directly or view it on GitHub
>>> <https://github.com/noirbizarre/flask-restplus/pull/85#issuecomment-149994967>
>>> .
>>>
>>
>>
>>
>> --
>> Robert Glen
>>
>
>
>
> --
> Robert Glen
>



-- 
Robert Glen

noirbizarre commented:
Thanks, I will look into it this weekend and ensure that it's fixed before I release the 0.8.1

robertglen commented:
Hello Axel,

I noticed the issue from the attached testapi.tar.gz still occurs in the
latest 0.8.2.   I am still stuck at 0.7.2 because my entire project wont
work beyond that release.  Is this still on your radar?

On Fri, Oct 23, 2015 at 1:41 AM, Axel Haustant <notifications@github.com>
wrote:

> Thanks, I will look into it this weekend and ensure that it's fixed before
> I release the 0.8.1
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/noirbizarre/flask-restplus/pull/85#issuecomment-150514057>
> .
>



-- 
Robert Glen

noirbizarre commented:
Hello Robert,

I just released the 0.8.4 dropping `flask_restful.Resource`. In both 0.8.3 and 0.8.4, I worked on error handling and dropped the original `flask_restful.Api.error_handler()`. Can you give it a try ?

I don't have the attached testapi.tar.gz (I think github doesn't process mail attachments) so I can't investigate your issue. Do you mind publishing a gist or a sample project ?

robertglen commented:
https://drive.google.com/file/d/0B8jedh85r36tR1NBdTVleFVVOWM/view?usp=sharing

I'm not sure how to upload files into gist, so I've got the tarball
accessible via google drive.

I've included a requirements.txt file, basically just make sure your
flask_restplus is 0.8.2 or greater to expose the error.

It happens to work fine on 0.8.0.  ./manage.py runserver

should throw the error.

Thanks.

On Mon, Dec 7, 2015 at 2:01 AM, Axel Haustant <notifications@github.com>
wrote:

> Hello Robert,
>
> I just released the 0.8.4 dropping flask_restful.Resource. In both 0.8.3
> and 0.8.4, I worked on error handling and dropped the original
> flask_restful.Api.error_handler(). Can you give it a try ?
>
> I don't have the attached testapi.tar.gz (I think github doesn't process
> mail attachments) so I can't investigate your issue. Do you mind publishing
> a gist or a sample project ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/noirbizarre/flask-restplus/pull/85#issuecomment-162468779>
> .
>



-- 
Robert Glen

noirbizarre commented:
OK, I found your error: you're redefining the resource constructor without taking `*args, **kwargs`.

You should write:
```python
@ns.route('/test', endpoint='dummy_path')
class DummyResource(Resource):

    def __init__(self, *args, **kwargs):
        self.reqparse = reqparse.RequestParser()
        self.reqparse.add_argument('arg_a', type=str, required=False,
                                   help='arg_a help')
        self.reqparse.add_argument('arg_b', type=str, required=False,
                                   help='arg_b help')
        super(DummyResource, self).__init__(*args, **kwargs)
```

robertglen commented:
Thanks, I'm sure that would have been a problem.  Turns out the other issue
I had was the way I was configuring the doc endpoint which changed in 0.8.2

I was instantiating all my Api with ui=False  then doing the previously
prescribed route decorator for the 'doc' endpoint and a call to
apidoc.ui_for.

Starting in 0.8.2 there is no more ui kwarg and the method is apparently
much simpler with the doc kwarg path.  Buy I was still registering via
decorator so the doc endpoint was being registered twice and clobbering.

I've changed my code and also added the resource *args, **kwargs
everywhere.  Thanks for looking into this.

Have a good holiday

-Robert.
On Dec 10, 2015 8:33 AM, "Axel Haustant" <notifications@github.com> wrote:

> OK, I found your error: you're redefining the resource constructor without
> taking *args, **kwargs.
>
> You should write:
>
> @ns.route('/test', endpoint='dummy_path')class DummyResource(Resource):
>
>     def __init__(self, *args, **kwargs):
>         self.reqparse = reqparse.RequestParser()
>         self.reqparse.add_argument('arg_a', type=str, required=False,
>                                    help='arg_a help')
>         self.reqparse.add_argument('arg_b', type=str, required=False,
>                                    help='arg_b help')
>         super(DummyResource, self).__init__(*args, **kwargs)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/noirbizarre/flask-restplus/pull/85#issuecomment-163679952>
> .
>
